opte/src/engine/packet.rs

```diff
@@ -1241,6 +1247,47 @@ impl Packet<Initialized> {
     }
 }
 
+/// A packet body transformation.
+///
+/// A body transformation allows an action to modify zero, one, or
+/// more bytes of a packet's body. The body starts directly after the
+/// ULP header, and continues to the last byte of the packet. This
+/// transformation is currently limited to only modifying bytes; it
+/// does not allow adding or removing bytes (e.g. to encrypt the body).
+pub trait BodyTransform: fmt::Display + DynClone {
+    /// Execute the body transformation. The body segments include
+    /// **only** body data, starting directly after the end of the ULP
+    /// header.
+    ///
+    /// # Errors
+    ///
+    /// The transformation can choose to return a
+    /// [`BodyTransformError`] at any time if the body is not
+    /// acceptable. On error, none or some of the bytes may have been
```

Comment:
Is there an argument to be made for running the transform on a copy of the data, and only swapping it into the original if the transform succeeds?

Reply:
OPTE should avoid copying the body. It's okay that the modification status is unknown on failure, the packet is going to be dropped anyways.

Reply:
Yeah, I know we want to avoid making copies, I was asking if there was a reason not to muck with the original body. Your point about it being dropped is a solid "no", so thanks!